src/internal_modules/roc_sndio/pump.cpp

```diff
@@ -72,6 +72,8 @@ status::StatusCode Pump::run() {
         code = status::StatusOK; // EOF is fine
     }
 
+    code = close_all_devices_();
```

Comment:
If code was not StatusOK before closing, we should return original code. Only if there were no previous errors before close, we should return code from close.

Currently, we overwrite code with return value of close. If close succeeds, we may overwrite previous error with StatusOK, which is bad. If close failed, we may overwrite previous with close error, which is not as bad, but also not desired - it's better to return earlier error because subsequent errors may be really caused by the first one.